src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PowCoinviewRule.cs

```diff
@@ -1,102 +1,72 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 using NBitcoin;
-using NBitcoin.Crypto;
 using Stratis.Bitcoin.Base.Deployments;
-using Stratis.Bitcoin.Features.Consensus.Interfaces;
-using Stratis.Bitcoin.Features.Consensus.Rules.CommonRules;
 using Stratis.Bitcoin.Utilities;
 
-namespace Stratis.Bitcoin.Features.Consensus
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
 {
     /// <summary>
-    /// Provides functionality for verifying validity of PoW block.
+    /// The proof of work coinview update rules. Validates the UTXO set is correctly spent and creating new outputs.
     /// </summary>
-    /// <remarks>PoW blocks are not accepted after block with height <see cref="Consensus.LastPOWBlock"/>.</remarks>
-    public class PowConsensusValidator : IPowConsensusValidator
+    /// <exception cref="ConsensusErrors.BadTransactionMissingInput">Thrown if transaction tries to spend inputs that are missing.</exception>
+    /// <exception cref="ConsensusErrors.BadTransactionNonFinal">Thrown if transaction's height or time is lower then provided by SequenceLock for this block.</exception>
+    /// <exception cref="ConsensusErrors.BadBlockSigOps">Thrown if signature operation cost is greater then maximum block signature operation cost.</exception>
+    /// <exception cref="ConsensusErrors.BadTransactionScriptError">Thrown if not all inputs are valid (no double spends, scripts & sigs, amounts).</exception>
+    [ExecutionRule]
+    public class PowCoinviewRule : ConsensusRule
     {
-        /// <summary>Flags that determine how transaction should be validated in non-consensus code.</summary>
-        public static Transaction.LockTimeFlags StandardLocktimeVerifyFlags = Transaction.LockTimeFlags.VerifySequence | Transaction.LockTimeFlags.MedianTimePast;
-
-        /// <summary>Instance logger.</summary>
-        private readonly ILogger logger;
-
-        /// <summary>Provider of block header hash checkpoints.</summary>
-        protected readonly ICheckpoints Checkpoints;
-
         /// <summary>Consensus parameters.</summary>
-        public NBitcoin.Consensus ConsensusParams { get; }
+        public NBitcoin.Consensus ConsensusParams { get; private set; }
 
         /// <summary>Consensus options.</summary>
-        public PowConsensusOptions ConsensusOptions { get; }
-
-        /// <summary>Keeps track of how much time different actions took to execute and how many times they were executed.</summary>
-        public ConsensusPerformanceCounter PerformanceCounter { get; }
+        public PowConsensusOptions ConsensusOptions { get; private set; }
```

Comment:
Again, I think this should be private

Reply:
And so should be the consensus parameters above

Reply:
I thought you just meant private set which is what I did, but yes it can be made a fully private member, as only the inheriting PosCoinviewRule uses it and that can get it direct from the Parent so I'll change it to do that.